chain/network/src/peer/peer_actor.rs

```diff
@@ -1383,28 +1417,18 @@ impl actix::Actor for PeerActor {
             }
             Some(reason) => {
                 tracing::info!(target: "network", "{:?}: Peer {} disconnected, reason: {reason}", self.my_node_info.id, self.peer_info);
-                let allow_reconnect = match reason {
-                    ClosingReason::TooManyInbound => false, // too many inbound
-                    ClosingReason::OutboundNotAllowed(_) => true,
-                    ClosingReason::Ban(_) => false, // banned
-                    ClosingReason::HandshakeFailed => true,
-                    ClosingReason::RejectedByPeerManager(_) => true,
-                    ClosingReason::StreamError => true,
-                    ClosingReason::DisallowedMessage => false, // misbehaving peer
-                    ClosingReason::PeerManagerRequest => false, // closed intentionally
-                    ClosingReason::DisconnectMessage => true,
-                    ClosingReason::TooLargeClockSkew => true,
-                    ClosingReason::OwnedAccountMismatch => false, // misbehaving peer
-                    ClosingReason::Unknown => true,
-                };
 
-                if !allow_reconnect {
+                if !reason.allow_reconnect_outbound() {
                     if let Some(other_peer) = self.other_peer_id() {
-                        self.network_state.remove_from_recent_outbound_connections(other_peer);
+                        self.network_state
+                            .connection_store
+                            .remove_from_recent_outbound_connections(other_peer);
                     }
                 }
 
-                self.send_message_or_log(&PeerMessage::Disconnect(Disconnect { allow_reconnect }));
+                self.send_message_or_log(&PeerMessage::Disconnect(Disconnect {
```

Comment:
I'm not sure if we discussed this, but it is kind of important that it is the outbound end of the connection which should try to reconnect (i.e. just one end of the connection should be responsible for reestablishing the connection). While here you apply the same logic no matter if the node is inbound/outbound end of the connection.